src/command_modules/azure-cli-batch/azure/cli/command_modules/batch/_client_factory.py

```diff
@@ -73,6 +73,14 @@ def batch_data_service_factory(cli_ctx, kwargs):
     account_endpoint = kwargs.pop('account_endpoint', None)
     kwargs.pop('yes', None)
 
+    # Verify all values are populated and display readable error
+    if not account_name:
+        raise ValueError("Account name not set.")
+    if not account_key:
+        raise ValueError("Account key not set.")
+    if not account_endpoint:
+        raise ValueError("Account endpoint not set.")
```

Comment:
This would be very annoying if say, none of these are set.  Then I will have to run the command three times and get a new error each time. Rather, it would be good to maintain a list of "missing_requireds" and then issue one error that identifies all missing parameters.

Reply:
Something like:
```Python
if not all(account_name, account_key, account_endpoint):
  raise ValueError('usage error: --account-name NAME --account-key KEY --account-endpoint ENDPOINT')
```

This is the "usage string format" that is encouraged throughout CLI and the user won't have to keep rerunning the command to find out the next missing thing.

Reply:
That will work.